### DISCUSSION TITLE
Support `DateOnly` and `TimeOnly` by default for .NET 6+

### DISCUSSION THREAD

**Discussion by SiCoe** [AutoFixture/AutoFixture#1304] [Ideas]
.NET 6.0 introduces two new types in the `System` namespace: [`DateOnly`](https://docs.microsoft.com/en-us/dotnet/api/system.dateonly) and [`TimeOnly`](https://docs.microsoft.com/en-us/dotnet/api/system.timeonly?view=net-6.0).

Due to how similar these are to [`DateTime`](https://docs.microsoft.com/en-us/dotnet/api/system.datetime), I propose we include these two new types to AutoFixtures set of [DefaultPrimitiveBuilders](https://github.com/AutoFixture/AutoFixture/blob/master/Src/AutoFixture/DefaultPrimitiveBuilders.cs) to allow AutoFixture to create instances without consumers needing to write additional code.

**Comment by SiCoe** [comment]
I may be able to spare some time to create a pull request for this change if maintainers are happy with the idea.

**Reply by aivascu** [reply]
I believe this would be a welcome addition to AutoFixture. Make sure to open the PR on the vnext branch `release/5.0.0`.

**Comment by SiCoe** [comment]
I've had a chance to look at this but would like maintainer feedback as my current thoughts are to add `net6.0` as a TargetFramework.

Please see pull request #1305 for my current thoughts.

I'd thought it best to raise this here as I'm aware the current [CONTRIBUTING](https://github.com/AutoFixture/AutoFixture/blob/master/CONTRIBUTING.md) guide states

> AutoFixture is currently being developed in C# on .NET 4.5.2 and .NET Standard using Visual Studio 2017 Update 5 or later

and this proposal will change that.

Do you think an additional target is a suitable approach, or would you favour something else / not implementing this feature?

**Reply by aivascu** [reply]
@SiCoe it is fine to add `net6.0`. This is exactly why the PR had to be raised on the vnext branch.
The statement you're referencing in the contributing is a bit stale at this moment. It will be updated along with other things.
I'll have look at the PR when I'll get some spare time.
